scripts/validate_snapshots.py

```diff
@@ -0,0 +1,110 @@
+"""
+Command line utility to validate snapshots prior to indexing by Azul
+"""
+
+import argparse
+from collections import (
+    defaultdict,
+)
+import json
+import logging
+import sys
+
+from azul import (
+    config,
+)
+from azul.args import (
+    AzulArgumentHelpFormatter,
+)
+from azul.azulclient import (
+    AzulClient,
+)
+from azul.bigquery import (
+    backtick,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+from azul.terra import (
+    TDRSourceSpec,
+)
+
+log = logging.getLogger(__name__)
+configure_script_logging(log)
+
+
+def main(args):
+    joiner = '||'
+    decline_snapshots = list()
+
+    azul = AzulClient(num_workers=1)
+    sources_by_catalog = {
+        catalog: azul.catalog_sources(catalog)
+        for catalog in args.catalogs
+    }
+    for catalog, sources in sources_by_catalog.items():
+        tdr_plugin = azul.repository_plugin(catalog)
+        log.info('Checking for %r in catalog %s', joiner, catalog)
+        if args.snapshot is not None:
+            sources = [s for s in sources if args.snapshot in s]
+        for spec in sources:
+            log.info('Validating snapshot %s', spec)
+            source = TDRSourceSpec.parse(spec)
+            tables = tdr_plugin._full_table_name(source, 'INFORMATION_SCHEMA.COLUMNS')
+            query = f'SELECT table_name, column_name FROM {backtick(tables)}'
+            rows = tdr_plugin._run_sql(query)
+            matching_rows = defaultdict(list)
```

Comment:
```suggestion
            table_columns = defaultdict(list)
```
`matching_rows` is confusing since the query was for all rows.
Since the key is a table name, and value is list of columns, `table_columns` seems like a natural fit.

Reply:
Good point!